### PR TITLE
feat: add the storage pkg for stat task

### PR DESCRIPTION
**Pull request by chlins**
This pull request introduces functionality for calculating and managing piece lengths for storage tasks, along with utilities for validating and retrieving task information. It includes new implementations, constants, helper functions, and comprehensive unit tests.

### Piece Length Calculation and Utilities:
* Added `CalculatePieceLength` function in `pkg/storage/piece.go` to compute optimal piece lengths based on content length, constrained by minimum and maximum thresholds. This function uses a helper, `nextPowerOfTwo`, to align lengths to powers of two.
* Defined constants `MIN_PIECE_LENGTH`, `MAX_PIECE_LENGTH`, and `MAX_PIECE_COUNT` to standardize piece length constraints.

### Task Management Enhancements:
* Introduced `StatTask` in `pkg/storage/stat_task.go` to retrieve task information based on parameters like path, URL, content length, and filtered query parameters. This includes validation and default handling for optional parameters.
* Added support for filtered query parameters specific to protocols (e.g., S3, GCS, OSS) for task ID generation.

### Unit Tests:
* Created unit tests in `pkg/storage/piece_test.go` to validate `CalculatePieceLength` behavior across edge cases, including minimum and maximum constraints and power-of-two alignment.
* Added unit tests in `pkg/storage/stat_task_test.go` to verify `StatTask` functionality, including success cases, error handling for missing parameters, and custom query parameter filtering.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

**Comment by gaius-qi** [comment]
@chlins Please rebase main branch.

File: pkg/storage/stat_task.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+	}
+
+	// containerdFilteredQueryParams is the default filtered query params with containerd to generate the task id.
+	containerdFilteredQueryParams = []string{
+		"ns",
+	}
+)
+
+func init() {
+	defaultFilteredQueryParams = append(defaultFilteredQueryParams, s3FilteredQueryParams...)
```
Comment (gaius-qi) [inline pkg/storage/stat_task.go:89]: Use `slices.Concat()`.

File: pkg/storage/stat_task.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+import (
+	"errors"
+	"os"
+	"path/filepath"
+
+	"d7y.io/dragonfly/v2/pkg/idgen"
+)
+
+// defaultFilteredQueryParams is the default filtered query params to generate the task id.
+var defaultFilteredQueryParams []string
```
Comment (gaius-qi) [inline pkg/storage/stat_task.go:28]: Move to `idgen` package.

File: pkg/storage/stat_task.go [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+	return func(task *statTask) {
+		task.filteredQueryParams = filteredQueryParams
+	}
+}
+
+// StatTask stats the task by the given parameters.
+func StatTask(path, url string, opts ...StatTaskOption) (os.FileInfo, error) {
+	st := &statTask{
+		path: path,
+		url:  url,
```
Comment (gaius-qi) [inline pkg/storage/stat_task.go:151]: `filteredQueryParams: defaultFilteredQueryParams`

**Review by gaius-qi — CHANGES_REQUESTED** [review]

**Review by gaius-qi — APPROVED** [review]
LGTM

**Review by Liam-Zhao — APPROVED** [review]
lgtm
